### PR TITLE
Add Greenland shelf regions to ocean analysis tasks

### PR DESCRIPTION
**Pull request by cbegeman**
Adds a default config section for Greenland regional TS Diagrams, hovmoller plots, and regional profiles. All new plots are off by default.

The option to designate 'all' regions in a regionGroup was added to the regional ocean profiles task.

### CODE REVIEW

**Comment by cbegeman** [comment]
@alexolinhager Do you want to look over the default config options here and see if we should adjust any of the settings, particularly as it pertains to T, S, and rho?

**Comment by cbegeman** [comment]
To be merged after https://github.com/MPAS-Dev/geometric_features/pull/199

**Comment by cbegeman** [comment]
Example results are located here: https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.cbegeman/analysis/20231105.v3b01.piControl.chrysalis/gris_regions_0001-0050_climo_0001-0050/ocean/index.html

**Comment by cbegeman** [comment]
This PR has been extended to include a few more analysis types.

File: mpas_analysis/default.cfg
Comment (alexolinhager) [inline mpas_analysis/default.cfg:1780]:
```suggestion
# See "regionNames" in the greenlandRegions masks file in
```

File: mpas_analysis/polar_regions.cfg
Comment (alexolinhager) [inline mpas_analysis/polar_regions.cfg:54]:
```suggestion
## options related to plotting T/S diagrams of Greenland regions
```

File: mpas_analysis/polar_regions.cfg
Comment (alexolinhager) [inline mpas_analysis/polar_regions.cfg:280]: Should we impose a depth range of -1000 – 0 m here to be consistent with the TS diagrams? At least for the Matt's coupling work that's the depth range we mostly care about, but I'm not sure if there would be other E3SM folks interested in seeing deeper profiles off the Greenland shelf

**Review by alexolinhager — APPROVED** [review]
@cbegeman This looks great, thanks again for putting this together! I left a few very minor comments on PR, but looks good to merge to me at least

It looks like I don't have admin privileges for MPAS-Analysis so you may also need Xylar or someone else to review it before it'll allow you to merge

**Comment by xylar** [comment]
@cbegeman, the version of `geometric_features` with these regions (1.4.0) should now be available.  Could you update the constraint on `geometric_features` here (it occurs in 2 places I think)?

What would you like me to focus on in my review?  Code? Testing? both?

**Comment by cbegeman** [comment]
@xylar Does this commit look like what you had in mind? I haven't had a chance to re-test yet so if you have a chance to test that would be great.

**Comment by xylar** [comment]
@cbegeman, could you rebase off of develop? I think this is probably bringing in merge commits from main.

**Comment by xylar** [comment]
And, yes, the last commit looks perfect, thanks.

**Comment by cbegeman** [comment]
> @cbegeman, could you rebase off of develop? I think this is probably bringing in merge commits from main.

At the time at which you commented, I think I had already rebased off develop. Can you take a look when you have a chance and let me know if it still looks wrong to you?

**Comment by xylar** [comment]
I ran some tests an things ran great for a QUwISC240 run but failed on a QU480 mesh:
```
Plotting Potential Temperature time series vs. depth...
  Load ocean data...
  Read in depth...
analysis task hovmollerOceanRegions_plotHovmoller_potentialTemperature_ISMIP6_Greenland_Central_West_Shelf failed during run
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.xylar/mpas_work/MPAS-Analysis/add-greenland-regions/mpas_analysis/shared/analysis_task.py", line 322, in run
    self.run_task()
  File "/gpfs/fs1/home/ac.xylar/mpas_work/MPAS-Analysis/add-greenland-regions/mpas_analysis/ocean/plot_hovmoller_subtask.py", line 382, in run_task
    fig, _, suptitle = plot_vertical_section_comparison(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/fs1/home/ac.xylar/mpas_work/MPAS-Analysis/add-greenland-regions/mpas_analysis/shared/plot/vertical_section.py", line 407, in plot_vertical_section_comparison
    _, ax = plot_vertical_section(
            ^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/fs1/home/ac.xylar/mpas_work/MPAS-Analysis/add-greenland-regions/mpas_analysis/shared/plot/vertical_section.py", line 885, in plot_vertical_section
    maskedTriangulation, unmaskedTriangulation = _get_triangulation(
                                                 ^^^^^^^^^^^^^^^^^^^
  File "/gpfs/fs1/home/ac.xylar/mpas_work/MPAS-Analysis/add-greenland-regions/mpas_analysis/shared/plot/vertical_section.py", line 1128, in _get_triangulation
    triMask = np.zeros((nx, ny, 2), bool)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: negative dimensions are not allowed
```
I suspect the issue is just that some of the regions are completely empty and we need a check for that and some way to handle it.  I'm happy to figure this out and fix it, I just don't have time until maybe Tuesday or so.

**Review by xylar — COMMENTED** [review]
In the meantime, I build the documentation and ran my test suite:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xasay-davis/analysis_testing/chrysalis/add-greenland-regions/
I think things look really great here (except the QU480 issue above, which I'll fix)!  My only remaining suggestion would be to turn on this analysis for polar runs by default.

**Comment by cbegeman** [comment]
@xylar Thanks for the suggestions and testing! Glad to hear you support adding Greenland regions to polar_regions; I wasn't sure how many additional plots were acceptable.

Seems like we would want to address the empty regions issue in a separate PR but let's put this one on hold until we're sure that's the reason why QU480 fails.

File: mpas_analysis/polar_regions.cfg
Comment (cbegeman) [reply mpas_analysis/polar_regions.cfg:273]: @alexolinhager Since we do not yet have the capability to actually mask the off-shelf quantities as of this PR, I think we should just keep this as-is and then change the range with the mask itself.

Comment (alexolinhager) [reply mpas_analysis/polar_regions.cfg:273]: Okay that makes sense to me!

**Comment by cbegeman** [comment]
@xylar I think this is ready to merge when we figure out how to handle the empty-mask case. Let me know if you need help with that.

**Comment by xylar** [comment]
@cbegeman, sorry, lost track of that one!  I'll work on it today.

**Review by xylar — APPROVED** [review]
I was able to fix the remaining issue with hovmoller plots on the QU480 grid by just leaving the plots blank.  I'll make a follow-up PR to fix that.

**Pull request merged**
